cmd/minishift/cmd/start.go

```diff
@@ -108,7 +109,12 @@ For the latter see 'minishift config -h'.`,
 		Run: runStart,
 	}
 
+	// enable experimental flags
+	//startCmd.Flags().BoolVar(&future, "future", false, "Enable experimental features (unsupported)")
```

Comment:
> did not work... uncomment this and try...

Sure, in this form it won't work. You will need to handle this a bit earlier in root.go. And you most likely have too work with hiding the flag. In general though I am pretty sure it should work. In the worst case you need to look up the '--show-experimental-flags' value before Viper is full initialised and depending on its value add the experimental flags to the flag set. 

I guess it comes down to what we think is the better approach. The environment variable one or via a flag. I think both can be made to work.